carddav/server.go

```diff
@@ -439,3 +439,19 @@ func (b *backend) Copy(r *http.Request, dest *internal.Href, recursive, overwrit
 func (b *backend) Move(r *http.Request, dest *internal.Href, overwrite bool) (created bool, err error) {
 	panic("TODO")
 }
+
+func PreconditionError(err string) error {
```

Comment:
We should probably create a separate type for precondition error types, e.g.

```go
type PreconditionErrorType string

const (
    PreconditionErrorNoUIDConflict PreconditionErrorType = "no-uid-conflict"
)
```

to avoid potential typos in backends